src/javascript/app/App/Components/Elements/Calendar/calendar-header.jsx

```diff
@@ -42,20 +43,26 @@ const CalendarHeader = ({
     const is_next_month_disabled  = isPeriodDisabled(addMonths(moment_date, 1), 'month');
     const is_next_year_disabled   = isPeriodDisabled(addYears(moment_date, num_of_years), 'month');
     const is_select_year_disabled = isPeriodDisabled(moment_date.clone().year(end_of_decade), 'year') || disable_year_selector;
+    const should_hide_next_month  = is_next_month_disabled && hide_disabled_periods;
+    const should_hide_prev_month  = is_prev_month_disabled && hide_disabled_periods;
+    const should_hide_prev_year   = is_prev_year_disabled && hide_disabled_periods;
+    const should_hide_next_year   = is_next_year_disabled && hide_disabled_periods;
 
     return (
         <div className='calendar__header'>
             <CalendarButton
                 className={classNames('calendar__nav calendar__nav--prev-year', {
-                    'calendar__nav--disabled': is_prev_year_disabled,
+                    'calendar__nav--disabled' : is_prev_year_disabled,
+                    'calendar__nav--is-hidden': should_hide_prev_year,
                 })}
                 onClick={is_prev_year_disabled ? undefined : () => navigateTo(subYears(calendar_date, num_of_years))}
```

Comment:
@easteregg @msamprz i think you need to update the check for the `onClick` events on the calendar buttons too. otherwise, it'll still be clickable although it is hidden. this one should be changed to `should_hide_prev_year`

Reply:
@kellybinary All done. thanks!